### PR TITLE
Lw test build revision

### PR DESCRIPTION
**Pull request by webberlh**
Revises the build process for the test suite to allow for building the library without tests. If Catch2 is not found, the test suite will be disabled entirely. If it is found, the test binaries will only be built when needed by the `test` target, the `all` target will only build the library and CLI.

A few warts with this approach. First, if Catch2 is not found, trying to build the `test` target will result in a message that no tests were found which isn't very helpful for hinting to the user that they need to install Catch2. There is a warning that will be emitted during the initial setup, but this could be easily missed with the amount of output. I'm thinking this problem may need to be addressed in the documentation, making sure we're explicit about the Catch2 dependency for testing. Second, the `-j` flag in `make test -j4` won't be passed through to the underlying `cmake` build, so in order to keep the parallelism a `cmake`-style build needs to be done with the `--parallel` flag. `make test` will still work it's just slower, so again this is probably just something that will need to be called out in the documentation.

### CODE REVIEW

File: tools/CMakeLists.txt [hunk 1]
```diff
@@ -25,4 +25,6 @@
 #
 
 add_subdirectory(main)
-add_subdirectory(test)
+if(DEFINED CATCH2_FOUND)
```
Comment (nalinigans) [inline tools/CMakeLists.txt:28]: Should we introduce a cmake option `DISABLE_TESTING` set to a default of `OFF` in addition? We should document that users who do not need testing built should explicitly set this to `ON`.

Comment (webberlh) [reply tools/CMakeLists.txt:28]: I considered having a dedicated cmake option for toggling it. The main advantage I saw for that approach was for a developer who had Catch2 installed but didn't necessarily want to pay the price of having to build the test binaries just to install the library. The additional change to defer the building until the test target is actually invoked should take care of this use case though. Was there another situation you were thinking of were having the option would be helpful?

Comment (nalinigans) [reply tools/CMakeLists.txt:28]: Yeah, if the user does have Catch2, but are building say the R bindings only(we should allow for the explicit building of OmicsDS from within build.sh for example), we should allow a disable test option. This way, `make test` will just be a noop.

Comment (nalinigans) [reply tools/CMakeLists.txt:28]: Since the test binaries are really only built during `make test`, these changes are fine.

**Review by nalinigans — APPROVED** [review]
Since test binaries are being built only during `make test`, these changes are fine.

**Pull request merged**